components/ws-daemon/pkg/controller/workspace_controller.go

```diff
@@ -219,7 +221,30 @@ func (wsc *WorkspaceController) handleWorkspaceRunning(ctx context.Context, ws *
 	span, ctx := opentracing.StartSpanFromContext(ctx, "handleWorkspaceRunning")
 	defer tracing.FinishSpan(span, &err)
 
-	return ctrl.Result{}, wsc.operations.SetupWorkspace(ctx, ws.Name)
+	var imageInfo *workspacev1.WorkspaceImageInfo = nil
+	if ws.Status.ImageInfo == nil {
+		id, err := wsc.runtime.WaitForContainer(ctx, ws.Name)
```

Comment:
I wonder whether we should do this async, just to avoid the potential of blocking the workspaces start. I totally see why it _should_ not block it, but... :shrug:

Reply:
Yes, but whether the cache is actually filled is not guaranteed. The two event-loops are not in sync in any way.

Reply:
It _could_ lead to problems if this method blocks, because the container already failed, and effectively never returns. Not saying it's likely, but while load testing workspace starts the last couple of weeks, I saw all kind of weird errors modes due to ws-daemon being out-of-sync with anything. :see_no_evil: 

But let's see, might be just working. :+1: